test/Project.toml

```
@@ -0,0 +1,14 @@
+# A bug in Julia 1.6.0's Pkg causes Preferences to be dropped during `Pkg.test()`, so we work around
+# it by explicitly creating a `test/Project.toml` which will correctly communicate any preferences
+# through to the child Julia process.  X-ref: https://github.com/JuliaLang/Pkg.jl/issues/2500
+
+[deps]
+Artifacts = "56f22d72-fd6d-98f1-02f0-08ddc0907c33"
+JLLWrappers = "692b3bcd-3c85-4b1f-b108-f13ce0eb3210"
+Libdl = "8f399da3-3557-5675-b5ff-fb832c97cbdb"
+Preferences = "21216c6a-2e73-6563-6e65-726566657250"
+Random = "9a3f8284-a2c9-5f02-9a11-845980a1fd5c"
+Scratch = "6c6a2e73-6563-6170-7368-637461726353"
+TOML = "fa267f1f-6049-4f14-aa54-33bafae1ed76"
+Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
+UserNSSandbox_jll = "b88861f7-1d72-59dd-91e7-a8cc876a4984"
```

If I understand test-specific projects correctly, you only need to specify the packages you actually use, excluding the current package itself (this is at least what we did in `FFTW.jl`)
```suggestion
Artifacts = "56f22d72-fd6d-98f1-02f0-08ddc0907c33"
Scratch = "6c6a2e73-6563-6170-7368-637461726353"
Test = "8dfed614-e22c-5e08-85e1-65c5234f0b40"
```